Z/P release notes updates (#28224)
openshift/openshift-docs: release_notes/ocp-4-7-release-notes.adoc

* Worker nodes must run {op-system-first}.
* Persistent shared storage must be provisioned by using either NFS or other supported storage protocols
* Persistent non-shared storage must be provisioned using local storage, like iSCSI, FC, or using LSO with DASD, FCP, or EDEV/FBA.
* These features are available only for {product-title} on IBM Z for {product-version}:
** HyperPAV enabled on IBM System Z /LinuxONE for the virtual machines for FICON attached ECKD storage

[id="ocp-4-7-ibm-power"]
==== IBM Power Systems

With this release, IBM Power Systems are now compatible with {product-title} {product-version}. See xref:../installing/installing_ibm_power/installing-ibm-power.adoc#installing-ibm-power_installing-ibm-power[Installing a cluster on IBM Power Systems] or xref:../installing/installing_ibm_power/installing-restricted-networks-ibm-power.adoc#installing-restricted-networks-ibm-power_installing-restricted-networks-ibm-power[Installing a cluster on IBM Power Systems in a restricted network] for installation instructions.

[discrete]
===== Notable Enhancements

The following new features are supported on IBM Power Systems with {product-title} {product-version}:

* Multipathing

> * {product-title} Pipelines TP

Reviewer: ```suggestion
* OpenShift Pipelines TP
```

> * {product-title} Pipelines TP

Reviewer: @vikram-redhat, why don't we add the product names for the aligned products to the main attributes file again?

Reviewer 2: Pipelines is technically not an aligned product but a feature of OCP. Are you asking if we should have `{piplines-title}` (or similar) instead of `{product-title} Pipelines TP`?

> * {product-title} Service Mesh

Reviewer: ```suggestion
* Red Hat OpenShift Service Mesh
```

* OVN-Kubernetes with an initial installation of {product-title} 4.7
* Persistent storage using Fibre Channel
* Persistent storage using Raw Block
* 4K Disk Support

[discrete]
===== Supported Features

The following features are also supported on IBM Power Systems:

* Currently, four Operators are supported:
** Cluster-Logging-Operator
** Cluster-NFD-Operator
** Elastic Search-Operator
** Local Storage Operator

* {product-title} Do (odo)
* CodeReady Workspaces
* Persistent storage using iSCSI
* HostPath

[discrete]
===== Restrictions

Note the following restrictions for {product-title} on IBM Power Systems:

* The following {product-title} features are unsupported:
** {product-title} Metering
** {product-title} Serverless
** {product-title} Virtualization
** CodeReady Containers (CRC)

* Worker nodes must run {op-system-first}.

> * Persistent storage must be of the Filesystem mode using local volumes, Network File System (NFS), or Container Storage Interface (CSI)

Reviewer: ```suggestion
* Persistent storage must be of the Filesystem type that uses local volumes, Network File System (NFS), or Container Storage Interface (CSI)
```

[discrete]
===== Notable Enhancements

The following new features are supported on IBM Z and LinuxONE with {product-title} {product-version}:

* KVM on {op-system-base} 8.3 or later is supported as a hypervisor for user-provisioned installation of {product-title} {product-version} on IBM Z and LinuxONE.  See xref:../installing/installing_ibm_z/installing-ibm-z-kvm.adoc#installing-ibm-z-kvm[Installing a cluster with {op-system-base} KVM on IBM Z and LinuxONE] for installation instructions.
* Multipathing
* OpenShift Pipelines TP
* Red Hat OpenShift Service Mesh
* OVN-Kubernetes with an initial installation of {product-title} 4.7
* Persistent storage using Fibre Channel
* Persistent storage using Raw Block
* z/VM Emulated FBA devices on SCSI disks

[discrete]
===== Supported features

The following features are also supported on IBM Z and LinuxONE:

* CodeReady Workspaces

> * OpenShift Do (odo)

Reviewer 3: Please note, odo name is itself odo and Not OpenShift Do. If you want to embellish it, you can say Developer CLI - odo, but the official name of the product is 'odo'
Also, do we want to mention CodeReady Workspaces in OCP docs. AFAIK, technically they are separately published on the customer portal.

Reviewer 4: If both are correct I would vote for "Developer CLI - odo" 

Regarding CRW, CRC, etc. I think its important for the IBM customers to have an overview what's supported on P/Z in regards to OpenShift. Especially since this constantly changes. 
I had this discussion with @codyhoag in regards to OCS and he said yes add it to the RNs. (Taken it out again because OCS generally isn't supported on 4.7)

Reviewer 5: I think it would be helpful to include things like CRW/CRC here, especially since we're providing such a comprehensive list of supported features. This provides one place where a customer can find supported features for IBM Z and LinuxONE, instead of them being spread around piecemeal, requiring additional searching.

We typically try to keep most features' direct release notes out of the core OCP release notes if they're documented elsewhere, but mentioning cross-product support for an item included in the core OCP release should be fine.

Reviewer 3: Ack @codyhoag makes sense.

* Persistent storage using iSCSI
* Persistent storage using local volumes (Local Storage Operator)

[discrete]
===== Restrictions

Note the following restrictions for {product-title} on IBM Z and LinuxONE:

* {product-title} for IBM Z does not include the following Technology Preview features:
** Log forwarding
** Precision Time Protocol (PTP) hardware
** CSI volume snapshots

* The following {product-title} features are unsupported:
** OpenShift Virtualization

> ** CodeReady Containers (CRC)

Reviewer 3: Not sure if we want to mention this here. Published separately on the Customer portal and not part of OCP docs. Though related to OCP, cause the product helps you run OCP locally.

Reviewer 2: If they are supported, no matter where the docs are published, we should mention them.

Space precluded documenting all of the container images for this release in the advisory. See the following article for notes on the container images in this release:

link:https://access.redhat.com/solutions/<ARTICLE_ID>[{product-title} 4.7.0 container image list]

[id="ocp-4-7-ibm-z"]
==== IBM Z and LinuxONE

With this release, IBM Z and LinuxONE is now compatible with {product-title} {product-version}. See xref:../installing/installing_ibm_z/installing-ibm-z.adoc#installing-ibm-z[Installing a cluster on IBM Z and LinuxONE] for installation instructions.

With this release, KVM on  {op-system-base} 8.3 or later is supported as a hypervisor for user-provisioned installation of {product-title} {product-version} on IBM Z and LinuxONE.  See xref:../installing/installing_ibm_z/installing-ibm-z-kvm.adoc#installing-ibm-z-kvm[Installing a cluster with {op-system-base} KVM on IBM Z and LinuxONE] for installation instructions.

[discrete]
===== Restrictions

Note the following restrictions for {product-title} on IBM Z and LinuxONE:

> 

Reviewer 6: We need to add currently two restrictions:
Bug #1903383 prevents installation on IBM z15 and LinuxONE III. To circumvent this problem on z15 and LinuxONE III please add to the cmdline option dfltcc=off.

Bug #1899762 OCP Nodes installed on FCP based SCSI storage might see unexpected crash. Use DASD base storage for the guest usage till RHEL 8 bug is not closed.

Reviewer 4: @codyhoag I assume this should go in general "Known Issues" section correct? Or would it be better to have IBM Z known issues direclty here?

@ktania46 Please add Holgers restrictions like this (wherever Cody says they belong :-) :
* Currently, installation on IBM z15 and LinuxONE III fails. As a workaround, add the option `dfltcc=off` to `boot.parm` via the command line. ([BZ#1903382](https://bugzilla.redhat.com/show_bug.cgi?id=1903383))

* Currently, OCP nodes installed on FCP-based SCSI storage might crash. As a workaround, use DASD-based storage for the guest usage. ([BZ#1899762](https://bugzilla.redhat.com/show_bug.cgi?id=1899762))

Reviewer 4: @ktania46 Please also add this to the Known Issues:  
* When installing OpenShift Container Platform on IBM Z with KVM, the recommended memory of 8 GB for OCP nodes might not be sufficient. ([BZ#1908875](https://bugzilla.redhat.com/show_bug.cgi?id=1908875))

fyi: @holgwolf

* Persistent storage using iSCSI
* Persistent storage using local volumes (Local Storage Operator)
* OpenShift Do (odo)
* OpenShift Pipelines TP
* Persistent storage using Fibre Channel
* Raw Block
* z/VM Emulated FBA devices on SCSI disks
* Multi-path Support
* OVNKube

[id="ocp-4-7-ibm-power"]
==== IBM Power Systems

With this release, IBM Power Systems are now compatible with {product-title} {product-version}. See xref:../installing/installing_ibm_power/installing-ibm-power.adoc#installing-ibm-power_installing-ibm-power[Installing a cluster on IBM Power Systems] or xref:../installing/installing_ibm_power/installing-restricted-networks-ibm-power.adoc#installing-restricted-networks-ibm-power_installing-restricted-networks-ibm-power[Installing a cluster on IBM Power Systems in a restricted network].

[discrete]
===== Restrictions

Note the following restrictions for {product-title} on IBM Power Systems:

* {product-title} for IBM Power Systems does not include the following Technology Preview features:
+
** OpenShift Virtualization
** OpenShift Serverless (knative, FaaS integrations)

* The following {product-title} features are unsupported:

> ** Red Hat OpenShift Service Mesh (istio, jaeger, kiali)

Reviewer 7: Will need to check the new date for Service Mesh. If this is before the 4.7 GA, we should mark Service Mesh as supported.

Reviewer 4: @manojnkumar The plan is that Service Mesh will go GA with OCP 4.6. Therefore, I would leave it as unsupported right now. 

cc @tgargya

Reviewer 7: @SNiemann15 : Is the service mesh GA captured elsewhere in the documentation?

Reviewer 4: @manojnkumar Yes we have it in Tanias PR for Service Mesh: https://github.com/openshift/openshift-docs/pull/28313

** {product-title} Metering
** OpenShift Metering (Presto, Hive)

* Worker nodes must run {op-system-first}.
* Persistent storage must be of the `Filesystem` mode using local volumes, Network File System (NFS), OpenStack Cinder, or Container Storage Interface (CSI).
* Networking must use either DHCP or static addressing with Red Hat Openshift SDN.
* AdoptOpenJDK with OpenJ9
* User-provisioned infrastructure
* Special Resources Operator
* OpenShift Ansible Service Broker Operator (deprecated)
* dotNET on RHEL

[discrete]
===== Supported Features

* Currently, five Operators are supported:
** Cluster-Logging-Operator
** Cluster-NFD-Operator
** Elastic Search-Operator
** Local Storage Operator

> ** SRIOV operator

Reviewer 7: We can remove the SRIOV operator from the 4.7 list.

* User-provisioned infrastructure deployment scenario on bare-metal
* OpenShift Cluster Monitoring
* Node Tuning Operator
* OpenShift Jenkins
* OpenShift Logging
* OpenShift Do (odo)
* Machine Configuration Operator, which is used in installations with installer-provisioned infrastructure
* Node Feature Discovery Operator
* {product-title} core (CVO Operators)
* Installation program for clusters that use user-provisioned infrastructure
* OVS
* RHEL8 Based container support
* RHEL CoreOS
* Ansible Engine
* Red Hat Software Collections
* HostPath
* iSCSI
* 4k Disk support
* CodeReady Workspaces
* OVNKube
* Fibre Channel
* Raw Block

> * Multus Plugins (SR-IOV, IPVAN, Bridge with VLAN, Static IPAM)

Reviewer 7: We can remove the Multus line.

** Red Hat OpenShift Service Mesh
** CodeReady Containers (CRC)
** {product-title} Metering
** Multus CNI plug-in
** FIPS cryptography
** Encrypting data stored in etcd
** Automatic repair of damaged machines with machine health checking
** Tang mode disk encryption during {product-title} deployment
** {product-title} Serverless
** Helm command-line interface (CLI) tool
** Controlling overcommit and managing container density on nodes
** etcd cluster Operator
** CSI volume cloning
** NVMe

* Worker nodes must run {op-system-first}.
* Persistent shared storage must be provisioned by using either NFS or other supported storage protocols
* Persistent non-shared storage must be provisioned using local storage, like iSCSI, FC, or LSO with DASD/FCP.
* These features are available only for {product-title} on IBM Z for {product-version}:
** HyperPAV enabled on IBM System Z /LinuxONE for the virtual machines for FICON attached ECKD storage

[discrete]
===== Supported features

With this release, the following features are supported on IBM Z and LinuxONE:

* Persistent storage using iSCSI
* Persistent storage using local volumes (Local Storage Operator)
* OpenShift Do (odo)

> * OpenShift Pipelines TP

Reviewer 4: @ktania46 Please add: 
* OVNKube
to the list of supported features. Sorry forgot it in my input.

* Persistent storage using local volumes (Local Storage Operator)
* OpenShift Do (odo)
* OpenShift Pipelines TP
* Persistent storage using Fibre Channel
* Raw Block
* z/VM Emulated FBA devices on SCSI disks
* Multi-path Support
* OVNKube

[id="ocp-4-7-ibm-power"]
==== IBM Power Systems

With this release, IBM Power Systems are now compatible with {product-title} {product-version}. See xref:../installing/installing_ibm_power/installing-ibm-power.adoc#installing-ibm-power_installing-ibm-power[Installing a cluster on IBM Power Systems] or xref:../installing/installing_ibm_power/installing-restricted-networks-ibm-power.adoc#installing-restricted-networks-ibm-power_installing-restricted-networks-ibm-power[Installing a cluster on IBM Power Systems in a restricted network].

[discrete]
===== Restrictions

Note the following restrictions for {product-title} on IBM Power Systems:

* {product-title} for IBM Power Systems does not include the following Technology Preview features:
+
** OpenShift Virtualization
** OpenShift Serverless (knative, FaaS integrations)

* The following {product-title} features are unsupported:
** Red Hat OpenShift Service Mesh (istio, jaeger, kiali)

> ** CodeReady Workspaces

Reviewer 7: CRW did GA in 2020. So we can remove it from the unsupported release.

** {product-title} Metering

> ** SR-IOV CNI plug-in

Reviewer 7: SR-IOV is scheduled to be added in 4.7.

> ** Red Hat Single Sign-On

Reviewer 7: SSO GAed in 2020.

** OpenShift Metering (Presto, Hive)

* Worker nodes must run {op-system-first}.
* Persistent storage must be of the `Filesystem` mode using local volumes, Network File System (NFS), OpenStack Cinder, or Container Storage Interface (CSI).
* Networking must use either DHCP or static addressing with Red Hat Openshift SDN.
* AdoptOpenJDK with OpenJ9

> * Userer-provisioned infrastructure

Reviewer 7: Userer -> User

* {op-system} real time (RT) kernels are currently only supported on compute nodes, not control plane nodes. Compact clusters are not supported with RT kernels in {product-title} 4.7. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1887007[*BZ#1887007*])

* Currently, installation on IBM z15 and LinuxONE III fails. As a workaround, add the option `dfltcc=off` to `boot.parm` via the command line. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1903383[*BZ#1903382*])

* On power environment, upon the node reboot, the crio and kubelet service may fail to start on any of the master or worker nodes causing the pods to remain in `ContainerCreating` state, this issue has been sometimes observed in 4.7 release. This will be fixed in a future 4.7.z release. To work around this issue, refer the Bugzilla (link:https://bugzilla.redhat.com/show_bug.cgi?id=1922154[*BZ#1922154*])

> 

Author: @manojnkumar @arcprabh

Author: @manojnkumar is it okay ?
